elements/pf-back-to-top/test/pf-back-to-top.spec.ts

```diff
@@ -0,0 +1,189 @@
+          <div style="height: 2000px;"></div>
+          <pf-back-to-top href="#top" scrollable-selector="#top">Back to top</pf-back-to-top>
+        </div>
+      `);
+      element = container.querySelector('pf-back-to-top')!;
+      snapshot = await a11ySnapshot();
+
+      await allUpdates(element);
+    });
+
+    it('should be hidden on init', async function() {
+      const { children } = snapshot;
+      expect(children).to.be.undefined;
+    });
+
+    it('should be visible after scrolling 401px', async function() {
+      const scrollableElement = document.querySelector('#top')!;
+      scrollableElement.scrollTo({ top: 401, behavior: 'instant' });
+      await nextFrame();
+      await allUpdates(element);
+      snapshot = await a11ySnapshot();
+      expect(snapshot.children?.map(takeProps(['name', 'role']))).to.deep.equal([{ role: 'link', name: 'Back to top' }]);
+    });
+  });
+
+  describe('when no text is provided in slot or title attribute', function() {
+    let element: PfBackToTop;
+    let snapshot: A11yTreeSnapshot;
+
+    describe('as a link', function() {
+      beforeEach(async function() {
+        await setViewport({ width: 320, height: 640 });
+        window.scrollTo({ top: 0, behavior: 'instant' });
+        await nextFrame();
+        const container = await createFixture<PfBackToTop>(html`
+          <div id="top">
+            <div style="height: 2000px;"></div>
+            <pf-back-to-top href="#top"></pf-back-to-top>
+          </div>
+        `);
+        element = container.querySelector('pf-back-to-top')!;
+        await allUpdates(element);
+      });
+
+      it('should have a label of "Back to top"', async function() {
+        window.scrollTo({ top: 401, behavior: 'instant' });
+        await nextFrame();
+        await allUpdates(element);
+        snapshot = await a11ySnapshot();
+        expect(snapshot.children?.map(takeProps(['name', 'role']))).to.deep.equal([{ role: 'link', name: 'Back to top' }]);
+      });
+    });
+
+    describe('as a button', function() {
+      beforeEach(async function() {
+        await setViewport({ width: 320, height: 640 });
+        window.scrollTo({ top: 0, behavior: 'instant' });
+        await nextFrame();
+        const container = await createFixture<PfBackToTop>(html`
+          <div id="top">
+            <div style="height: 2000px;"></div>
+            <pf-back-to-top></pf-back-to-top>
+          </div>
+        `);
+        element = container.querySelector('pf-back-to-top')!;
+        await allUpdates(element);
+      });
+
+      it('should have a label of "Back to top"', async function() {
+        window.scrollTo({ top: 401, behavior: 'instant' });
+        await nextFrame();
+        await allUpdates(element);
+        snapshot = await a11ySnapshot();
+        expect(snapshot.children?.map(takeProps(['name', 'role']))).to.deep.equal([{ role: 'button', name: 'Back to top' }]);
+      });
+    });
+  });
+});
```

Comment:
Do we want to include some tests for the following?
- whether it is keyboard navigable
- whether there is an accessible label when there is nothing slotted and label property is blank